app/scripts/controllers/network/createInfuraClient.test.js

```diff
@@ -0,0 +1,232 @@
+    it('does not hit Infura, instead returning the chain id that maps to the Infura network, as a hex string', async () => {
+      const chainId = await withInfuraClient(
+        { network: 'ropsten' },
+        ({ makeRpcCall }) => {
+          return makeRpcCall({
+            method: 'eth_chainId',
+          });
+        },
+      );
+
+      expect(chainId).toStrictEqual('0x3');
+    });
+  });
+
+  describe('when the RPC method is net_version', () => {
+    it('does not hit Infura, instead returning the chain id that maps to the Infura network, as a decimal string', async () => {
+      const chainId = await withInfuraClient(
+        { network: 'ropsten' },
+        ({ makeRpcCall }) => {
+          return makeRpcCall({
+            method: 'net_version',
+          });
+        },
+      );
+
+      expect(chainId).toStrictEqual('3');
+    });
+  });
+
+  // == RPC methods that do not support params
+  //
+  // For `eth_getTransactionByHash` and `eth_getTransactionReceipt`, a
+  // "non-empty" result is not only one that is not null, undefined, or a
+  // geth-specific "nil", but additionally a result that has a `blockHash` that
+  // is not 0x0.
+
+  describe('eth_blockNumber', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_blockNumber');
+  });
+
+  describe('eth_compileLLL', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_compileLLL');
+  });
+
+  describe('eth_compileSerpent', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_compileSerpent');
+  });
+
+  describe('eth_compileSolidity', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_compileSolidity');
+  });
+
+  describe('eth_estimateGas', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_estimateGas');
+  });
+
+  describe('eth_gasPrice', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_gasPrice');
+  });
+
+  describe('eth_getBlockByHash', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_getBlockByHash');
+  });
+
+  describe('eth_getBlockTransactionCountByHash', () => {
+    testsForRpcMethodThatDoesNotSupportParams(
+      'eth_getBlockTransactionCountByHash',
+    );
+  });
+
+  describe('eth_getBlockTransactionCountByNumber', () => {
+    testsForRpcMethodThatDoesNotSupportParams(
+      'eth_getBlockTransactionCountByNumber',
+    );
+  });
+
+  describe('eth_getCompilers', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_getCompilers');
+  });
+
+  describe('eth_getFilterLogs', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_getFilterLogs');
+  });
+
+  describe('eth_getTransactionByBlockHashAndIndex', () => {
+    testsForRpcMethodThatDoesNotSupportParams(
+      'eth_getTransactionByBlockHashAndIndex',
+    );
+  });
+
+  describe('eth_getTransactionByBlockNumberAndIndex', () => {
+    testsForRpcMethodThatDoesNotSupportParams(
+      'eth_getTransactionByBlockNumberAndIndex',
+    );
+  });
+
+  describe('eth_getTransactionByHash', () => {
```

Comment:
Both of these two methods take a "transaction hash" parameter

Reply:
I still need to take care of this as well.

Reply:
I can resolve this thread now that the helper function was renamed. We still need to test the parameter caching behavior, but that's true of everything.